### ISSUE TITLE
[BUG] renders incorrectly in emacs vterm

### ISSUE THREAD

**Issue by smallstepman** [aristocratos/btop#64]
**Describe the bug**

displays incorrectly (see screenshots)

**To Reproduce**

1. install emacs 27
2. install vterm, the way I'm doing it is with [doom](https://github.com/hlissner/doom-emacs)
```
git clone --depth 1 https://github.com/hlissner/doom-emacs ~/.emacs.d
~/.emacs.d/bin/doom install
# open ~/.doom.d/init.el, uncomment line with `vterm` by deleting leading semicolons
~/.emacs.d/bin/doom sync
```
3. open emacs
4. run vterm - either with Meta-x shortcut and search for "+vterm/here", or, press [Space o t]
4. run btop

**Expected behavior**

render similarly to bpytop

**Screenshots**
#### btop
![image](https://user-images.githubusercontent.com/21069150/135712716-48d8061b-50aa-4a6f-90e5-fba05bb6ada1.png)
![image](https://user-images.githubusercontent.com/21069150/135712738-c9f7fd43-cfae-4c46-8a46-491fc5a71682.png)
![image](https://user-images.githubusercontent.com/21069150/135712731-7bd0f5b0-06a0-4878-9dca-db9a453c27c7.png)
#### bpytop 
![image](https://user-images.githubusercontent.com/21069150/135712780-1dc8ebee-1c13-457d-89be-646d34d1dc3f.png)
![image](https://user-images.githubusercontent.com/21069150/135712789-f7864365-3d9f-4a57-9699-cd173ef446d2.png)

**Info (please complete the following information):**
 - btop++ version: 1.0.12
 - Binary:  https://archlinux.org/packages/community/x86_64/btop/
 - Architecture: x86_64
 - Platform: Linux
 - (Linux) Kernel: 5.12.15-arch1-1
 - (OSX/FreeBSD) Os release version: not sure
 - Terminal used: https://github.com/akermu/emacs-libvterm
 - Font used: JetBrains Mono

**Additional context**

```
❯ cat ~/.config/btop/btop.log 

2021/10/02 (12:42:00) | ===> btop++ v.1.0.12
2021/10/02 (12:42:00) | DEBUG: Starting in DEBUG mode!
2021/10/02 (12:42:00) | INFO: Logger set to DEBUG
2021/10/02 (12:42:00) | DEBUG: Using locale en_US.UTF-8
2021/10/02 (12:42:00) | INFO: Running on /dev/pts/1
2021/10/02 (12:42:00) | INFO: Forcing tty mode: setting 16 color mode and using tty friendly graph symbols
2021/10/02 (12:42:06) | INFO: Quitting! Runtime: 00:00:06
```
I also tried flipping all setting in `general` tab, but it didn't help

**Comment by aristocratos** [comment]
@smallstepman 
I think this is related to the symbol used for empty space in the graphs, see #60 for specifics about the symbol.
So it's likely an issue with how libvterm handles the width of that symbol, you can try looking for vterm settings related to character width, but there's really nothing I can do about it.

**Comment by aristocratos** [comment]
@smallstepman 
Realized it was actually not issue to change to regular whitespace.
Fixed in [v1.0.13](https://github.com/aristocratos/btop/releases/tag/v1.0.13)

**Comment by k4r4b3y** [comment]
**I am having the same issue** as in the screenshots of @WuuBoLin .  I installed emacs, elpa-vterm, and btop packages from debian bookworm stable repositories:

- emacs: GNU Emacs 29.4
- elpa-vterm: Version: 0.0.2+git20230217.3e5a9b7-1
- btop: Version: 1.2.13-1

**Comment by JasZhe** [comment]
@k4r4b3y @WuuBoLin Not sure if ya'll fixed it but for others who might stumble upon this

That might be a font issue I had something similar. If you go into vterm-copy-mode, hover over one of the braille characters and do `C-u C-x =` (what-cursor-position w/prefix arg) you'll probably see something like "Apple Braile" font which _isn't_ monospaced which causes that alignment issue. What I did to fix it was to install a monospaced unicode font i.e. GNU unifont https://www.unifoundry.com/unifont/index.html
https://www.unifoundry.com/pub/unifont/unifont-17.0.03/font-builds/ (direct link to download)

Then set that font to be used for unicode-bmp `(set-fontset-font t 'unicode-bmp "Unifont")`

**Comment by WuuBoLin** [comment]
Hey @JasZhe, sorry for the late reply. _Yes, I did fix it._ 
**Indeed**, it does turn out to be a font issue, exactly as you described.

I recently realized this [while using OpenCode](https://github.com/sst/opencode/issues/431#issuecomment-3638210238) as well, and was able to confirm the root cause.
The solution you provided is actually a better permanent fix for the issue. (For those who still want to use their own font for most other unicode glyphs, my solution in the previous link might be better.)

At some point my smol brain suddenly went *“wait… isn’t this just a font issue because Emacs render text buffers?”* So I tried `C-u C-x =`, just like you suggested in evil normal mode.   But I completely forgot this issue that I had already commented on it before existed LOL. My bad!

Thanks for pointing this out!
